pkg/controller/decorator/controller.go

```diff
@@ -302,9 +302,15 @@ func (c *decoratorController) processNextWorkItem() bool {
 	defer c.queue.Done(key)
 
 	if err := c.sync(key.(string)); err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to sync %v '%v': %w", c.dc.Name, key, err))
-		c.queue.AddRateLimited(key)
-		return true
+		if apierrors.IsNotFound(err) {
```

Comment:
I think this is handled in https://github.com/metacontroller/metacontroller/blob/master/pkg/controller/decorator/controller.go#L480-L483 - if parent was not found error is not return, thus we do not enter here